gofor/index.js

```diff
@@ -51,19 +31,31 @@ module.exports = class Gofor {
         this.fetch = (...args) => {
             args[1] = this.setOptions(args[1]);
 
-            return fetch(...args);
+            return this.fetcher(...args);
         };
 
         this.fetch.config = this.config.bind(this);
     }
 
-    /**
-     * Convert self's literal headers to Headers when applicable
-     * no return value
-     */
-    convertHeaders() {
-        if (this[defaults] && this[defaults].headers) {
-            this[defaults].headers = toHeaders(this[defaults].headers);
+    get fetcher() {
+        return fetch;
+    }
+
+    get interfaces() {
+        return {
+            Headers,
+            Request,
+            Response
+        };
+    }
+
+    get supportsHeaders() {
+        try {
+            const { Headers } = this.interfaces;
+
+            return typeof Headers !== 'undefined' && Headers.prototype && typeof Headers.prototype.entries === 'function';
```

Comment:
Since this is already a try/catch block
```suggestion
            return typeof Headers.prototype.entries === 'function';
```